libyul/AsmParser.cpp

```diff
@@ -553,8 +553,7 @@ bool Parser::isValidNumberLiteral(string const& _literal)
 	try
 	{
 		// Try to convert _literal to u256.
-		auto tmp = u256(_literal);
-		(void) tmp;
+		[[maybe_unused]] auto tmp = u256(_literal);
```

Comment:
This should never be optimised out.

Reply:
``[[maybe_unused]]`` only tells the compiler not to warn. It'll never just optimize this out, if the right hand side can throw on some inputs.

Reply:
We could consider just using ``static_cast<u256>(_literal);`` instead, though - that might be easier on the eye...

Reply:
I think calling a constructor is easier on the eye than casting...

Reply:
All fine with me.